### PR TITLE
Masonry: add a11y roles

### PR DESCRIPTION
**Pull request by dangerismycat**
Masonry is a list of items, but we don't present it as such semantically. That's not great for accessibility.

As best I can tell, there are three rendering scenarios for Masonry. In two of them, we render some form of a container with items. In the remaining one we render an empty container. For the first two, this PR adds `role="list"` to the main container, and `role="listitem"` to the individual Pin items. No changes were made to the empty container div, since (I think) it doesn't need a role.

### CODE REVIEW

File: packages/gestalt/src/Masonry.js [hunk 1]
```diff
@@ -382,8 +384,7 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
     this.forceUpdate();
   }
 
-  // $FlowFixMe[unclear-type]
-  renderMasonryComponent: (itemData: T, idx: number, position: *) => Node = (
+  renderMasonryComponent: (itemData: T, idx: number, position: Position) => Node = (
```
Comment (dangerismycat) [inline packages/gestalt/src/Masonry.js:387]: unrelated Flow cleanup

File: packages/gestalt/src/Masonry.js [hunk 2]
```diff
@@ -541,7 +545,7 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
         ? Math.max(...positions.map((pos) => pos.top + pos.height))
         : 0;
       gridBody = (
-        <div style={{ width: '100%' }} ref={this.setGridWrapperRef}>
+        <div style={{ width: '100%' }} ref={this.setGridWrapperRef} role="list">
           <div className={styles.Masonry} style={{ height, width }}>
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Masonry.js:549]: I believe the role="list" needs to be at this level. With the screenreader, I'm hearing "List 1 item" because it's only seeing this div, whereas when I move it, I get "list 15 items"

**Review by ayeshakmaz — CHANGES_REQUESTED** [review]

File: packages/gestalt/src/Masonry.js (hunk 2 above)
Comment (dangerismycat) [reply packages/gestalt/src/Masonry.js]: you are absolutely correct! 🤦

**Review by liuyenwei — APPROVED** [review]

**Review by ayeshakmaz — APPROVED** [review]
